### PR TITLE
Help Article: Comparing columns through raw query

### PR DESCRIPTION
**Pull request by nurul3101**
Help Article Candidate: Comparing columns of the same table through raw queries.

## Describe this PR

Added Help article which can be added at: https://www.prisma.io/docs/support/help-articles

## Any other relevant information

Internal Notion Document: [Comparing columns help article](https://www.notion.so/prismaio/Comparing-Columns-of-Same-table-through-Raw-Queries-861442d9ac814d2c8d2d0d8c02979686)

### CODE REVIEW

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 1]
```diff
@@ -0,0 +1,193 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing columns of same table through raw queries'
+---
+
+## Workaround
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
## Issue
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+Comparing columns from a same table is a common scenario which developers have to deal with. Some examples of it could be comparing two numeric values in a same table or comparing two dates in a same table.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table.
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+
+## Solution
+
+This could be achieved by using [Raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Since we never want to start a sentence with "This" or "It" or anything that is non-specific, let's reword this to something like: "Comparing values from two columns in the same table can be achieved by using raw queries..."

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+A use case could be of getting posts which has more comments than likes, In this case we need to compare commentsCount and likesCount.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
One use case for comparing values from different columns would be retrieving posts that have more comments than likes; in this case we need to compare the values of `commentsCount` and `likesCount`.
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  content       String?
+  published     Boolean  @default(false)
+  author        User     @relation(fields: [authorId], references: [id])
+  authorId      Int
+  likesCount    Int
+  commentsCount Int
+}
+```
+
+Queries depending upon databases could look something like:
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Queries (depending upon which database) could look something like:
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+  const response =
+    await prisma.$queryRaw`SELECT * FROM "Post" WHERE "likesCount" < "commentsCount";`;
+
+  console.log(response);
+}
+
+await initiateNumbersComparisonRawQuery();
+```
+
+Executing the above queries depending upon the database would filter posts which has less likes compared to comments.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: There's a guideline in tech writing to avoid using the word "execute" unless absolutely needed. In. this case, we can say "Running the above queries...." 

Also, in this sentence we want "that" not "which" (because the clause is important to the entire meaning).

And, we need to put in parentheses the clause "depending upon the database".... and finally, use "fewer" instead of "less" (weird grammar rule about if you can count them, use "fewer").

So final edited sentence would be:
"Running the above queries (depending upon the database) would filter posts that has fewer likes compared to comments.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+    published: false,
+    authorId: 1,
+    likesCount: 50,
+    commentsCount: 100
+  }
+]
+```
+---
+
+Similarly If we need to compare dates we could achieve the same thing using raw queries as well.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Similarly, if we need to compare dates we could achieve the same thing using raw queries as well.
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+A use case could be to get all the projects which are completed before the due date.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
A use case could be to get all the projects that are completed before the due date.
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+  title         String
+  author        User     @relation(fields: [authorId], references: [id])
+  authorId      Int
+  dueDate       DateTime
+  completedDate DateTime
+  createdAt     DateTime @default(now())
+}
+```
+
+Queries depending upon databases could look something like:
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Queries (depending upon the database) could look something like:
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+  const response =
+    await prisma.$queryRaw`SELECT * FROM \`public\`.\`Project\` WHERE \`completedDate\` > \`dueDate\`;`;
+
+  console.log(response);
+}
+
+await initiateDatesComparisonRawQuery();
+```
+
+Executing the above query would filter projects whose completion date is after it’s due date.
```
Comment (tanberry) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Running the above query would filter projects whose completion date is after its due date.
```

**Review by tanberry — COMMENTED** [review]
Thanks @nurul3101 ! Let me know if any questions, and after you've made the suggested changes either me or Lucy can merge it.

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks @nurul3101 ! Let me know if any questions, and after you've made the suggested changes either me or Lucy can merge it.

**Review by tanberry — CHANGES_REQUESTED** [review]
Thanks @nurul3101 ! Let me know if any questions, and after you've made the suggested changes either me or Lucy can merge it.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 3 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Updated the sentence

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 6 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Thanks Tana for the amazing suggestions.
I have updated the sentences.
I think we are good to go. All comments are resolved.

**Comment by nurul3101** [comment]
All suggested edits are done @tanberry. It looks like we are good to merge.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 11]
```diff
@@ -0,0 +1,193 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing columns of same table through raw queries'
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: These two should probably be different.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+## Issue
+
+Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table.
+
+
+## Solution
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: As mentioned somewhere else, not a "Solution" but a "Workaround" for missing Prisma functionality that is tracked in an issue. If we call it "Solution" people might misunderstand that we are not planning to actually implement the solution.

Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Updated the heading from "Solution" to "Workaround"

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 11 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Updated the `metaDescription` content. Thanks for pointing that out.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+---
+
+## Issue
+
+Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table.
+
+
+## Workaround
+
+Comparing values from two columns in the same table can be achieved by using [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Comparing values from two columns in the same table can be achieved by using [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 14]
```diff
@@ -0,0 +1,193 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing column values through prisma's inbuilt raw query methods'
+---
+
+## Issue
+
+Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table.
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Maybe add the link to the issue that tracks this as a proper feature of Prisma Client?

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+    title: 'Hello World',
+    content: 'This is my first post',
+    published: false,
+    authorId: 1,
+    likesCount: 50,
+    commentsCount: 100
+  }
+]
+```
+---
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: I don't think we ever use horizontal lines like this. If a page needs structure, we usually add headlines that describe the content that follows.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 14 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Added the GitHub Issue Link - https://github.com/prisma/prisma/issues/5048

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 15 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Removed the horizontal divider. Thanks for letting me know 👍

Comment (janpio) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: I think giving both of the examples a headline would work nice.

Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Sure, I have added headlines to both examples.
- Comparing numeric values
- Comparing date values

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 16]
```diff
@@ -0,0 +1,194 @@
...
+Comparing different columns from the same table is a common scenario that developers encounter. Some examples include comparing two numeric values in the same table or comparing two dates in a same table. There's an existing [GitHub Issue](https://github.com/prisma/prisma/issues/5048) regarding the same. 
+
+
+## Workaround
+
+Comparing values from two columns in the same table can be achieved by using [raw queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).
+
+### Comparing numeric values
+
+One use case for comparing values from different columns would be retrieving posts that have more comments than likes; in this case we need to compare the values of `commentsCount` and `likesCount`.
```
Comment (TasinIshmam) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
One use case for comparing values from different columns would be retrieving posts that have more comments than likes; in this case, you need to compare the values of `commentsCount` and `likesCount`.
```

The [docs style guide](https://www.prisma.io/docs/about/prisma-docs/style-guide#use-you-as-the-pronoun) suggests using the second person ("you") instead of first person ("we").

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+    authorId: 1,
+    likesCount: 50,
+    commentsCount: 100
+  }
+]
+```
+
+### Comparing date values
+
+Similarly, if we need to compare dates we could achieve the same thing using raw queries as well.
```
Comment (TasinIshmam) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Similarly, if you need to compare dates, you could also achieve the same thing using raw queries.
```

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 18]
```diff
@@ -0,0 +1,194 @@
...
+
+A use case could be to get all the projects that are completed before the due date.
```
Comment (TasinIshmam) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
For example, a use case could be to get all projects completed after the due date.
```

The query seems to be fetching projects where `completedDate` > `dueDate`. Would this not imply you're getting projects completed *after* the due date?

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 19]
```diff
@@ -0,0 +1,194 @@
...
+  const response =
+    await prisma.$queryRaw`SELECT * FROM \`public\`.\`Project\` WHERE \`completedDate\` > \`dueDate\`;`;
+
+  console.log(response);
+}
+
+await initiateDatesComparisonRawQuery();
+```
+
+Running the above query would filter projects whose completion date is after its due date.
```
Comment (TasinIshmam) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]:
```suggestion
Running the above query would fetch projects where `completedDate` is after the `dueDate`. 
```

Nitpick: makes it a bit more explicit what the query is doing.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 16 above)
Comment (TasinIshmam) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx:18]: The [docs style guide](https://www.prisma.io/docs/about/prisma-docs/style-guide#use-you-as-the-pronoun) suggests using the second person ("you") instead of first person ("we").

**Review by TasinIshmam — COMMENTED** [review]
Hey @nurul3101 great work with the article!

I left a few suggestions that you might consider.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx (hunk 18 above)
Comment (nurul3101) [reply content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: Thanks for catching this Tasin! 👍

**Comment by nurul3101** [comment]
Thanks @TasinIshmam for suggestions 👍 
I have applied the edits.

File: content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx [hunk 20]
```diff
@@ -0,0 +1,194 @@
+---
+title: 'Comparing columns of same table through raw queries'
+metaTitle: 'Comparing columns of same table through raw queries'
+metaDescription: 'Comparing columns with prisma's inbuilt raw query methods'
```
Comment (janpio) [inline content/500-support/100-help-articles/500-comparing-columns-through-raw-queries.mdx]: This should fix the build error:
```suggestion
metaDescription: 'Comparing columns with Prisma\'s inbuilt raw query methods'
```

**Review by tanberry — APPROVED** [review]
Looks good, thanks!
